### PR TITLE
feat: replaces code block with Kongponents component

### LINKED ISSUE #400: move codeblock to kongponents
### Description

This would be useful to many and it makes sense to upstream it

### PR DESCRIPTION
**Pull request by kleinfreund**
## Changes

Replaces most of the implementation of `CodeBlock` with the new Kongponents component `KCodeBlock`.

Resolves #400.

Signed-off-by: Philipp Rudloff <philipp.rudloff@konghq.com>

## Screenshots

image

### CODE REVIEW

**Review by lahabana — APPROVED** [review]

File: src/components/CodeBlock.spec.ts [hunk 1]
```diff
@@ -5,10 +5,7 @@ import CodeBlock from './CodeBlock.vue'
 function renderComponent(props = {}) {
```
[the remark this answers is not in the document]
Comment (kleinfreund) [reply src/components/CodeBlock.spec.ts:5]: There isn’t really a good argument to keep them, no. The component is test thoroughly in Kongponents and the syntax highlighting in Kuma GUI. The tests also pass with the new component so I can throw them out. I had them adjusted before the component was available in Kongponents to check if I perhaps run into any edge cases which wasn’t so.

[the remark this answers is not in the document]
Comment (kleinfreund) [reply src/components/CodeBlock.spec.ts]: I’ve removed them.
